libsolidity/codegen/ir/Common.cpp

```diff
@@ -135,10 +135,9 @@ string dispenseLocationComment(langutil::SourceLocation const& _location, IRGene
 {
 	solAssert(_location.sourceName, "");
 	_context.markSourceUsed(*_location.sourceName);
-	return AsmPrinter::formatSourceLocationComment(
+	return "/// " + AsmPrinter::formatSourceLocationComment(
```

Comment:
This is intentional, the ast-id is printed only for specific solidity ast nodes and is done in the IRGenerator directly. Or did I misunderstand something here?

Reply:
But it means that if you read the IR and then print it again with `AsmPrinter`, you won't ever see `@ast-id`s, even after you start parsing them (unless of course you change this part in the PR that adds parsing but I see no reason not to do that already here).

Reply:
You're right. it's fine.

Reply:
> I think that because of this `--ir` will at least have much fewer `@ast-id` tags than `--ir-optimized`.

With this I assumed that the AST ID would be printed whenever `@src` gets printed. But now I realize that it will only be there if it was explicitly added to debug data.